### PR TITLE
Auto active LocationComponent while style loaded.

### PR DESCRIPTION
**Pull request by Chaoba**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: #323 #324 

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>Enable LocationComponent automatically when style loaded; fix null island location puck</changelog>`.

### Summary of changes

- The root cause for location not shown when accepting permissions is we need to get the style to active location component 
and the style has not been set yet while init location component.
This pr listen to the style load event and trigger `applySettings ` to reactive location component and apply a loop request to get the location update after location permission is granted.

- This pr also make the location puck invisible before getting the real location from the device to fix the null island issue

### CODE REVIEW

File: plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationProviderImpl.kt [hunk 1]
```diff
@@ -28,9 +29,11 @@ internal class LocationProviderImpl(context: Context) : LocationProvider, Locati
 
   @SuppressLint("MissingPermission")
   private fun requestLocationUpdates() {
-    locationEngine.requestLocationUpdates(
-      locationEngineRequest, this, Looper.getMainLooper()
-    )
+    if (PermissionsManager.areLocationPermissionsGranted(context)) {
```
Comment (pengdev) [inline plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationProviderImpl.kt:32]: shall we add some error log when the permission is not granted at this point of time?

File: plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationProviderImpl.kt [hunk 2]
```diff
@@ -76,7 +79,9 @@ internal class LocationProviderImpl(context: Context) : LocationProvider, Locati
       requestLocationUpdates()
     }
     locationConsumers.add(locationConsumer)
-    locationEngine.getLastLocation(this)
+    if (PermissionsManager.areLocationPermissionsGranted(context)) {
```
Comment (pengdev) [inline plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationProviderImpl.kt:82]: same as above

**Review by pengdev — APPROVED** [review]
Great catch, LGTM % comments

**Review by kiryldz — APPROVED** [review]

File: plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationProviderImpl.kt (hunk 1 above)
Comment (Chaoba) [reply plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationProviderImpl.kt:32]: Yes, will do.

File: plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationProviderImpl.kt [hunk 3]
```diff
@@ -26,11 +29,26 @@ internal class LocationProviderImpl(context: Context) : LocationProvider, Locati
...
+      )
+    } else {
+      if (handler == null) {
+        handler = Handler()
+        runnable = Runnable { requestLocationUpdates() }
+      }
+      handler?.postDelayed(runnable, UPDATE_DELAY)
+      Logger.w(
+        TAG,
+        "Missing location permission, location component will not take effect before location permission is granted."
```
Comment (tobrun) [inline plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationProviderImpl.kt:49]: I'm ok with a loop that verifies if permissions have been expected but 2 remarks:
 - shouldn't we clean up the handler when the location/map is destroyed (we could end up with a loop that continuous forever)
 - can we apply an exponential backoff + decrease the initial UPDATE_DELAY so it eventually grows to the current value?

Comment (Chaoba) [reply plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationProviderImpl.kt:49]: The runnable will be removed in `unRegisterLocationConsumer` which would be invoked while deactivateLocationComponent

**Review by tobrun — APPROVED** [review]

**Pull request merged**